ols/app/endpoints/ols.py

```diff
@@ -245,19 +244,15 @@ def retrieve_attachments(llm_request: LLMRequest) -> list[Attachment]:
 def generate_response(
     conversation_id: str,
     llm_request: LLMRequest,
-    previous_input: list[dict[Literal["type", "content"], str]],
+    previous_input: list[CacheEntry],
 ) -> SummarizerResponse:
     """Generate response based on validation result, previous input, and model output."""
     # Summarize documentation
     try:
         docs_summarizer = DocsSummarizer(
             provider=llm_request.provider, model=llm_request.model
         )
-        history = [
-            conversation["type"] + ": " + conversation["content"].strip()
-            for conversation in previous_input
-            if conversation
-        ]
+        history = CacheEntry.cache_entries_to_history(previous_input)
```

Comment:
@onmete Previously we were removing the whitespaces from the "content" - dont see that happening with this changes - may be I wam missing to see ?

Reply:
Right, I've added that to `cache_entries_to_history` and extra test for that.